openedx/core/djangoapps/content/block_structure/signals.py

```diff
@@ -23,6 +23,7 @@ def _update_block_structure_on_course_publish(sender, course_key, **kwargs):  #
     if isinstance(course_key, LibraryLocator):
         return
 
+    # TODO: flagging this line for further investigation. Is this flag on currently? Should it be unconditional?
     if config.waffle().is_enabled(config.INVALIDATE_CACHE_ON_PUBLISH):
         clear_course_from_cache(course_key)
```

Comment:
Also, I believe that this is trying to clear an lms cache from cms, which is pointless. Note the below method call, [which uses the cross-process queueing system](https://github.com/edx/edx-platform/blob/master/cms/celery.py#L39)